helios-services/src/test/java/com/spotify/helios/agent/TaskRunnerTest.java

```diff
@@ -113,4 +129,49 @@ public void testPullTimeoutVariation() throws Throwable {
       assertEquals(ImagePullFailedException.class, t.getCause().getClass());
     }
   }
+
+  @Test
+  public void testContainerNotRunningVariation() throws Throwable {
+
+    final ContainerInfo stopped = new ContainerInfo() {
+      @Override
+      public ContainerState state() {
+        final ContainerState state = mock(ContainerState.class);
+        when(state.running()).thenReturn(false);
+        when(state.error()).thenReturn("container is a potato");
+        return state;
+      }
+    };
+
+    doReturn("potato").when(mockCreation).id();
```

Comment:
I saw `doThrow` in the test above and used tab completion to test whether there was a similar `doReturn`. I then noticed the `when().thenReturn()` pattern in another test file. So not really. Is there a reason to prefer one or the other?